crocks/List.js

```diff
@@ -27,6 +27,13 @@ const _of =
 const _empty =
   () => List([])
 
+function fromArray(xs) {
+  if(!arguments.length || !isArray(xs)) {
+    throw new TypeError('List.fromArray: Must wrap an array')
+  }
+  return List(xs)
```

Comment:
We will not be able to use this on the same issue this work was defined.

So instead we will have to use a reduction here to get back a`List`.

As `List` is a `Monoid` we could probably use `mconcat` here like this:

```js
function fromArray(xs) {
  if(!arguments.length || !isArray(xs)) {
    throw new TypeError('List.fromArray: Must wrap an array')
  }
  return mconcat(List, xs)
```